electronics/src/Config.h

```diff
@@ -0,0 +1,103 @@
+#ifndef CONFIG_H
+#define CONFIG_H
+
+#include "Handles.h"
+#include <PinNames.h>
+
+/**
+ *
+ * -----------------------------------------------
+ * SET MODE [Type of controller to be deployed to]
+ * -----------------------------------------------
+ * 
+ */
+
+// #define RAM // Red Alliance controller mode
+// #define BAM // Blue Alliance controller mode
+// #define SGM // Shield Generator controller mode
+#define STM // Scoring Table controller mode
+
+#ifdef RAM
+#define MODE 0
+#endif
+
+#ifdef BAM
+#define MODE 1
+#endif
+
+#ifdef SGM
+#define MODE 2
+#endif
+
+#ifdef STM
+#define MODE 3
+#endif
+
+#define global static const // Constant experssion accessable to all (cannot be modified)
+
+/**
+ * Config for Power port
+ */
+#ifdef RAM
+#define IP "10.0.100.4"
```

Comment:
IP for all devices should be given by DHCP, not static. Devices will report to the FMS and advertise their "role" in the first message

Reply:
Let's also look at the possibility of setting the "mode"/"role" at runtime - we can store it in flash like in my example which would avoid needing to reflash controllers every time we setup.

Reply:
Ip and port stuff is mostly just a placeholder, we should probably go through what we actually want to do with communications. 

E.g TCP or UDP, standard buffer size? maybe 256? etc...
I'm not completely sure what networking you're imagining, so soon we need to go over some kind of network map, and how we want everything being done